test/distributed/_tensor/test_math_ops.py

```diff
@@ -126,6 +126,43 @@ def test_softmax_with_bwd(self):
             self.assertIsNotNone(dist_x.grad)
             self.assertEqual(dist_x.grad.full_tensor(), x.grad)
 
+    @with_comms
+    def test_nll_loss_fwd(self):
+        device_mesh = self.build_device_mesh()
+
+        channel_size, channel_dim = 16, 1
+        test_setup = [
+            (2, (8, channel_size), (8,)),  # calling aten.nll_loss_forward
+            (3, (8, channel_size, 12), (8, 12)),  # calling aten.nll_loss2d_forward
+        ]
+        for input_ndim, input_size, target_size in test_setup:
+            x = torch.rand(*input_size, device=self.device_type)
+            target = torch.randint(channel_size, target_size, device=self.device_type)
+            dist_target = distribute_tensor(target, device_mesh, [Replicate()])
+
+            shard_dims = list(range(input_ndim))
+            reductions = ["none", "mean", "sum"]
+            for shard_dim, reduction in itertools.product(shard_dims, reductions):
+                y = torch.nn.functional.nll_loss(x, target, reduction=reduction)
+                dist_x = distribute_tensor(x, device_mesh, [Shard(shard_dim)])
+                dist_y = torch.nn.functional.nll_loss(
+                    dist_x, dist_target, reduction=reduction
+                )
+                if shard_dim == channel_dim:
+                    self.assertTrue(dist_y.placements[0].is_replicate())
```

Comment:
add `comm_debug` test?

Reply:
currently `comm_debug` doesn't count communications happening in sharding prop, probably because the __torch_dispatch__ at op level overrides that at the `comm_debug` level; left a TODO item to fix it